django_sharding_library/fields.py

```diff
@@ -156,3 +159,30 @@ class ShardForeignKeyStorageField(ShardForeignKeyStorageFieldMixin, ForeignKey):
     the shard using a pre_save signal.
     """
     pass
+
+
+class PostgresShardGeneratedIDField(AutoField):
+    """
+    A field that uses a Postgres stored procedure to return an ID generated on the database.
+    """
+    def db_type(self, connection, *args, **kwargs):
+
+        if not hasattr(settings, 'SHARD_EPOCH'):
+            raise ValueError("PostgresShardGeneratedIDField requires a SHARD_EPOCH to be defined in your settings file.")
+
+        if connection.vendor == PostgresDatabaseWrapper.vendor:
+            return "bigint DEFAULT next_sharded_id()"
+        else:
+            return super(PostgresShardGeneratedIDField, self).db_type(connection)
+
+    @staticmethod
+    def migration_receiver(*args, **kwargs):
+        sequence_name = "global_id_sequence"
```

Comment:
Correct, its per DB. There is no reason to have a per model, no DB (that I am aware of) can insert more than 1024 rows per millisecond (which is what this id generator is designed to be able to do)!